python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -328,38 +328,62 @@ def dependencies(self) -> DependencyMapping[NodeInvocation]:
     @public
     @property
     def executor_def(self) -> ExecutorDefinition:
+        """Returns the default :py:class:`ExecutorDefinition` for the job.
```

Comment:
IMO this would benefit from a note on overriding the "default", because the next paragraph makes it sound like the"default" is just `multi_or_in_process_executor`. Maybe:

>Returns the default :py:class:`ExecutorDefinition` for the job. This is the `ExecutorDefinition` that will be used if one is not specified at runtime via config.

Reply:
I agree with the sentiment, although not entirely accurate to say that the default executor `multi_or_in_process_executor` comes from `Definitions` object, since it will be set on the job if none is provided (just overwritten)